### PR TITLE
Simplify+optimize

### PR DESCRIPTION
**Pull request by mralusw**
Hi, I've found `deb-get` really useful and played with it for a bit.

I've tried to optimize performance (by avoiding repeated external calls and caching some results &mdash; e.g. `list` is about 4 times faster), fix some (minor) bugs and simplify some shell constructs (e.g. replacing `grep | cut` with `sed ...`), fix some `shellcheck` warnings etc.

I've tried to make the diffs between commits as small / clear as possible (as opposed to a big rewrite), hence the number of commits.

I've tested it as much as I could on my normal workflows, hopefully github-actions can also help here (I'm not sure how to run testing on my fork).

### CODE REVIEW

**Comment by mralusw** [comment]
Hm, I got Error: Resource not accessible by integration... Not sure what that means

**Comment by philclifford** [comment]
Thanks for this excellent and instructive work. Don't worry about the error you mentioned: the actions seem to be broken lately and need some attention and revision.
I've tested this PR lightly with normal update/upgrade cycles on Ubuntu-Mate (with no problems other than normal transient upstream issues).  I'd like to get time to run some github-actions testing across a small matrix of distros but that work (and testing and fixing the tests) is beyond my available bandwidth for a while.
To  run github-actions on your fork you just need to enable it in the settings of the fork , but first you might want to tweak the actions definitions in your fork/branch and also possibly ensure those tweaks don't accidentally get added to your PR.

File: deb-get [hunk 1]
```diff
@@ -471,18 +440,17 @@ function validate_deb() {
             # If the method is github and the cache file is empty, ignore the package
             # The GitHub API is rate limit has likely been reached
             if [ "${METHOD}" == github ] && [ ! -s "${CACHE_FILE}" ]; then
-                if [ "${ACTION}" != "prettylist" ] && [ "${ACTION}" != "remove" ] && [ "${ACTION}" != "purge" ]; then
+                if ! [[ ' prettylist remove purge ' =~ " ${ACTION} " ]]; then
```
Comment (philclifford) [inline deb-get:443]: :eyes:

**Review by philclifford — DISMISSED** [review]
LGTM 
Limited operational testing so far

**Comment by philclifford** [comment]
Looking at things I rarely use part 1: `search`

```
(base) ✔ ~/src/deb-get [official|…4⚑ 2] 
03:29 $ ./deb-get search "tri"
lutris
motrix
tribler
trilium
trivy
(base) ✔ ~/src/deb-get [official|…4⚑ 2] 
03:30 $ ./deb-get search "^tri"
tribler
trilium
trivy
(base) ✔ ~/src/deb-get [official|…4⚑ 2] 
03:30 $ git checkout simplify+optimize 
Switched to branch 'simplify+optimize'
(base) ✔ ~/src/deb-get [simplify+optimize L|…4⚑ 2] 
03:30 $ ./deb-get search "^tri"
1password
agena
android-messages-desktop
antimicrox
...
zoom
zotero

```

**Pull request merged**